File: solutions/scala/134-move-zeroes.scala
/**
 * Problem 134: Move Zeroes (LeetCode 283)
 * Difficulty: Easy
 * Language: Scala
 */
def moveZeroes(nums: List[Int]): List[Int] = {
  val (zeros, nonZeros) = nums.partition(_ == 0)
  nonZeros ++ zeros
}

File: solutions/typescript/134-move-zeroes.ts
/**
 * Problem 134: Move Zeroes (LeetCode 283)
 * Difficulty: Easy
 * Language: TypeScript
 */
function moveZeroes(nums: number[]): number[] {
  const nonZeros = nums.filter(x => x !== 0);
  const zeros = nums.filter(x => x === 0);
  return [...nonZeros, ...zeros];
}
